migrations/20170101000000_initial.js

```diff
@@ -0,0 +1,49 @@
+exports.up = (knex, Promise) => {
+  const schoolsTable = knex.schema.createTable('schools', (table) => {
+    table.increments('id');
+    table.string('name').notNullable().unique();
+    table.string('abbreviation', 32);
+  });
+
+  const departmentsTable = knex.schema.createTable('departments', (table) => {
+    table.uuid('id').notNullable().primary();
+    table
+      .integer('school_id')
+      .notNullable()
+      .references('id')
+      .inTable('schools')
+      .onDelete('CASCADE')
+      .onUpdate('CASCADE');
+    table.string('name').notNullable();
+    table.unique(['school_id', 'name']);
+  });
+
+  const venuesTable = knex.schema.createTable('venues', (table) => {
+    table
+      .integer('school_id')
+      .notNullable()
+      .references('id')
+      .inTable('schools')
+      .onDelete('CASCADE')
+      .onUpdate('CASCADE');
+    table.string('name').notNullable();
+    table.string('type');
+    table.string('owned_by');
+    table.primary(['school_id', 'name']);
```

Comment:
I would strongly recommend against using a composite index as primary key. For instance, it would be difficult to add a new table that need to reference this (say, a table for module lessons). 

You can set up an additional index for this instead if you're concerned about query speed, but the PK should be a single column only.